### ISSUE TITLE
Error when loading in julia 1.6.0-rc1

### ISSUE THREAD

**Issue by gbaraldi** [SciML/ReservoirComputing.jl#88]
Calling `Using ReservoirComputing` gives the following errors 
```julia
julia> using ReservoirComputing
ERROR: InitError: UndefVarError: Params not defined
Stacktrace:
  [1] top-level scope
    @ none:1
  [2] eval
    @ ./boot.jl:360 [inlined]
  [3] define_typeinf_code2()
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/reflection/reflection.jl:21
  [4] __init__()
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/IRTools.jl:37
  [5] _include_from_serialized(path::String, depmods::Vector{Any})
    @ Base ./loading.jl:670
  [6] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:756
  [7] _tryrequire_from_serialized(modkey::Base.PkgId, build_id::UInt64, modpath::String)
    @ Base ./loading.jl:685
  [8] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:745
  [9] _tryrequire_from_serialized(modkey::Base.PkgId, build_id::UInt64, modpath::String)
    @ Base ./loading.jl:685
 [10] _require_search_from_serializedjulia> using ReservoirComputing
ERROR: InitError: UndefVarError: Params not defined
Stacktrace:
  [1] top-level scope
    @ none:1
  [2] eval
    @ ./boot.jl:360 [inlined]
  [3] define_typeinf_code2()
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/reflection/reflection.jl:21
  [4] __init__()
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/IRTools.jl:37
  [5] _include_from_serialized(path::String, depmods::Vector{Any})
    @ Base ./loading.jl:670
  [6] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:756
  [7] _tryrequire_from_serialized(modkey::Base.PkgId, build_id::UInt64, modpath::String)
    @ Base ./loading.jl:685
  [8] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:745(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:745
 [11] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:994
 [12] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:910
 [13] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:897
during initialization of module Inner

julia> using ReservoirComputing
ERROR: InitError: LoadError: KeyError: key %9 not found
Stacktrace:
  [1] getindex
    @ ./dict.jl:482 [inlined]
  [2] substitute(p::IRTools.Inner.Pipe, x::Union{IRTools.Inner.NewVariable, IRTools.Inner.Variable})
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/ir/ir.jl:853
  [3] _broadcast_getindex_evalf
    @ ./broadcast.jl:648 [inlined]
  [4] _broadcast_getindex
    @ ./broadcast.jl:621 [inlined]
  [5] getindex
    @ ./broadcast.jl:575 [inlined]
  [6] copyto_nonleaf!(dest::Vector{Zygote.Alpha}, bc::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{1}, Tuple{Base.OneTo{Int64}}, typeof(IRTools.Inner.substitute), Tuple{Tuple{IRTools.Inner.Pipe}, Base.Broadcast.Extruded{Vector{Any}, Tuple{Bool}, Tuple{Int64}}}}, iter::Base.OneTo{Int64}, state::Int64, count::Int64)
    @ Base.Broadcast ./broadcast.jl:1078
  [7] copy
    @ ./broadcast.jl:930 [inlined]
  [8] materialize
    @ ./broadcast.jl:883 [inlined]
  [9] substitute(p::IRTools.Inner.Pipe, x::Expr)
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/ir/ir.jl:856
 [10] substitute(p::IRTools.Inner.Pipe, x::IRTools.Inner.Statement)
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/ir/ir.jl:855
 [11] iterate(p::IRTools.Inner.Pipe, ::Any)
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/ir/ir.jl:892
 [12] renumber(ir::Any)
    @ IRTools.Inner ~/.julia/packages/IRTools/BpoqK/src/passes/passes.jl:124
 [13] |>
    @ ./operators.jl:859 [inlined]
 [14] Zygote.Adjoint(ir::IRTools.Inner.IR; varargs::Nothing, normalise::Bool)
    @ Zygote ~/.julia/packages/Zygote/KNUTW/src/compiler/reverse.jl:287
 [15] top-level scope
    @ ~/.julia/packages/Zygote/KNUTW/src/precompile.jl:15
 [16] include(mod::Module, _path::String)
    @ Base ./Base.jl:386
 [17] include
    @ ~/.julia/packages/Zygote/KNUTW/src/Zygote.jl:1 [inlined]
 [18] precompile
    @ ~/.julia/packages/Zygote/KNUTW/src/Zygote.jl:46 [inlined]
 [19] (::Zygote.var"#1837#1838")()
    @ Zygote ~/.julia/packages/Requires/035xH/src/init.jl:11
 [20] __init__()
    @ Zygote ~/.julia/packages/Requires/035xH/src/init.jl:18
 [21] _include_from_serialized(path::String, depmods::Vector{Any})
    @ Base ./loading.jl:670
 [22] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:756
 [23] _tryrequire_from_serialized(modkey::Base.PkgId, build_id::UInt64, modpath::String)
    @ Base ./loading.jl:685
 [24] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:745
 [25] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:994
 [26] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:910
 [27] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:897
in expression starting at /home/gabrielbaraldi/.julia/packages/Zygote/KNUTW/src/precompile.jl:15
during initialization of module Zygote
```

If I call `using ReservoirComputing` for a third time it just works.

This is the version I am using
``` julia> versioninfo()
Julia Version 1.6.0-rc1
Commit a58bdd9010 (2021-02-06 15:49 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
```

**Comment by MartinuzziFrancesco** [comment]
This seems to be the same error that occurred in https://github.com/JuliaRegistries/General/pull/30964 that @ChrisRackauckas mentioned to be caused by a IRTools issue

**Comment by ChrisRackauckas** [comment]
Yes it's an upstream issue with the Zygote/Flux stack.

**Comment by MartinuzziFrancesco** [comment]
This has been solved since the stable release of Julia 1.6. I'll close the issue
